### PR TITLE
common: assertion with --bind and invalid ip address

### PR DESCRIPTION
**Pull request by shinobu-x**
fix: http://tracker.ceph.com/issues/22358

Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by wjwithagen** [comment]
So how about IPv6??

And why do all this work, when the socket routines will tell you that it is an invalid address??

File: src/common/config.cc [hunk 1]
```diff
@@ -480,6 +481,11 @@ int md_config_t::parse_argv(std::vector<const char*>& args)
       set_val_or_die("mon_host", val.c_str());
     }
     else if (ceph_argparse_witharg(args, i, &val, "--bind", (char*)NULL)) {
+      static const boost::regex e("\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}");
+      if (!boost::regex_match(val.c_str(), e)) {
+        std::cerr << "error: invalid ip address " << val.c_str() << std::endl;
+        _exit(1); 
+      }
       set_val_or_die("public_addr", val.c_str());
```
Comment (shinobu-x) [inline src/common/config.cc]: @wjwithagen it's up to what's expected value for `public_addr`.

**Comment by tchaikov** [comment]
retest this please

**Comment by shinobu-x** [comment]
@tchaikov thanks for fix!!

**Comment by shinobu-x** [comment]
test this please

**Comment by tchaikov** [comment]
retest this please.

**Comment by liewegas** [comment]
I think this belongs in teh config value checker for the address type (for the bind config option), and then switch to something other than set_val_or_die so we can return a friendly error

**Pull request closed without merging**